src/main/java/com/synacy/gradprogram/spock/exercise/RefundService.java

```diff
@@ -4,14 +4,36 @@
 
 public class RefundService {
 
-  public BigDecimal calculateRefund() {
-    // TODO: Implement me. Full refund if cancel reason is due to damaged item.
-    //  Also full refund if the order was cancelled within 3 days of order date, else refund half of the total cost.
-    return null;
+  private final RefundRepository refundRepository;
+
+  public RefundService(RefundRepository refundRepository) {
+    this.refundRepository = refundRepository;
+  }
+
+  public BigDecimal calculateRefund(CancelOrderRequest request, Order order) {
+    long refundDateLimit = order.getDateOrdered().getTime() + (86400000 * 3);
```

Comment:
I understand what we are doing here but I believe there are other ways to do this. This kinda makes things hard to understand. We should try to have our code easy to understand and read :+1: I would suggest figuring out a more elegant way of doing this logic :+1: